consensus-types/blocks/roblob.go

```diff
@@ -0,0 +1,42 @@
+package blocks
+
+import (
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+)
+
+var errNilBlockHeader = errors.New("received nil beacon block header")
+
+// ROBlob represents a read-only blob sidecar with its block root.
+type ROBlob struct {
+	*ethpb.BlobSidecar
+	root [32]byte
+}
+
+// NewROBlobWithRoot creates a new ROBlob with a given root.
+func NewROBlobWithRoot(b *ethpb.BlobSidecar, root [32]byte) (ROBlob, error) {
+	if b == nil {
+		return ROBlob{}, errNilBlock
+	}
+	return ROBlob{BlobSidecar: b, root: root}, nil
+}
+
+// NewROBlob creates a new ROBlob by computing the HashTreeRoot of the header.
+func NewROBlob(b *ethpb.BlobSidecar) (ROBlob, error) {
+	if b == nil {
+		return ROBlob{}, errNilBlock
+	}
+	if b.SignedBlockHeader == nil || b.SignedBlockHeader.Header == nil {
+		return ROBlob{}, errNilBlockHeader
+	}
+	root, err := b.SignedBlockHeader.Header.HashTreeRoot()
+	if err != nil {
+		return ROBlob{}, err
+	}
+	return ROBlob{BlobSidecar: b, root: root}, nil
+}
+
+// BlockRoot returns the root of the block.
+func (b *ROBlob) BlockRoot() [32]byte {
```

Comment:
I added a `RootSlice` helper to the `ROBlock` package, that could be helpful here to streamline the transition in places where we want to move from the protobuf type `BlockRoot` value to `BlockRoot`, in the instance where we actually do want a slice and not a `[32]byte`. The reason I did this for ROBlock is that in places where the caller actually wants a slice, it can be more concise to do the slicing in the helper func, otherwise the caller has to make a temporary variable in order to slice it (because you can't slice a function return).

Reply:
not a PR blocker btw, we can add this if it turns out to be helpful when we start porting over the usages of `.BlockRoot`.

Reply:
sure! I'll add one now